examples/gaussian_process/plot_gpr_noisy_targets.py

```diff
@@ -11,104 +11,143 @@
+gaussian_process = GaussianProcessRegressor(kernel=kernel, n_restarts_optimizer=9)
+gaussian_process.fit(X_train, y_train)
+gaussian_process.kernel_
+
+# %%
+# After fitting our model, we see that the hyperparameters of the kernel have
+# been optimized. Now, we will use our kernel to compute the mean prediction
+# of the full dataset and plot the 95% confidence interval.
+mean_prediction, std_prediction = gaussian_process.predict(X, return_std=True)
+
+plt.plot(X, y, label=r"$f(x) = x \sin(x)$", linestyle="dotted")
+plt.scatter(X_train, y_train, label="Observations")
+plt.plot(X, mean_prediction, label="Mean prediction")
+plt.fill_between(
+    X.ravel(),
+    mean_prediction - 1.96 * std_prediction,
+    mean_prediction + 1.96 * std_prediction,
+    alpha=0.5,
+    label=r"95% confidence interval",
+)
+plt.legend()
+plt.xlabel("$x$")
+plt.ylabel("$f(x)$")
+_ = plt.title("Gaussian process regression on noise-free dataset")
+
+# %%
+# We see that for a prediction close to a training sample, the 95% confidence
+# interval is small. Whenever a sample is falls far from training data, our
+# model's prediction is less accurate and its confidence interval is larger.
+#
+# Example with noisy targets
+# --------------------------
+#
+# We can repeat a similar experiment but this time we will add an additional
+# noise to the target. It will allow us to see the effect of the noise on the
+# fitted model.
+#
+# We define the noise standard deviation and add it to the original training
+# target.
+dy = 0.5 + 1.0 * rng.random_sample(y_train.shape)
+y_train_noisy = y_train + rng.normal(0, dy)
+
+# %%
+# We create a similar Gaussian process model. In addition to the kernel, this
+# time, we specify the parameter `alpha` that is related to the variance of the
+# noise.
+gaussian_process = GaussianProcessRegressor(
+    kernel=kernel, alpha=dy ** 2, n_restarts_optimizer=9
+)
+gaussian_process.fit(X_train, y_train_noisy)
+mean_prediction, std_prediction = gaussian_process.predict(X, return_std=True)
+
+# %%
+# Let's plot the mean prediction and the confidence interval as before.
+plt.plot(X, y, label=r"$f(x) = x \sin(x)$", linestyle="dotted")
+plt.errorbar(
+    X_train,
+    y_train,
+    dy,
+    linestyle="None",
+    color="tab:blue",
+    marker=".",
+    markersize=10,
+    label="Observations",
+)
+plt.plot(X, mean_prediction, label="Mean prediction")
+plt.fill_between(
+    X.ravel(),
+    mean_prediction - 1.96 * std_prediction,
+    mean_prediction + 1.96 * std_prediction,
+    color="tab:orange",
+    alpha=0.5,
+    label=r"95% confidence interval",
+)
+plt.legend()
+plt.xlabel("$x$")
+plt.ylabel("$f(x)$")
+_ = plt.title("Gaussian process regression on a noisy dataset")
+
+# %%
+# The noise affect the predictions close to the training samples: now, the
+# confidence interval close to the training samples is larger because we
+# indicated the amount of noise for each training samples.
```

Comment:
```suggestion
# The noise affects the predictions close to the training samples: the
# predictive uncertainty near to the training samples is larger because we
# explicitly model a given level target noise independent of the input
# variable.
```